tests/test_lane.py

```diff
@@ -74,6 +75,101 @@ def apply(self, qu):
         f.apply(_db, qu)
         eq_(qu, ep.called_with)
 
+    def test_from_request(self):
+        # from_request just calls _from_request.
+        expect = object()
+        @classmethod
+        def mock(cls, facet_config, get_argument, worklist, **extra_kwargs):
+            return expect
+        old = FacetsWithEntryPoint._from_request
+        FacetsWithEntryPoint._from_request = mock
+        eq_(expect, FacetsWithEntryPoint.from_request(None, None, None, None))
+        FacetsWithEntryPoint._from_request = old
```

Comment:
This test breaks the next one because `FacetsWithEntryPoint._from_request` isn't restored exactly how it was before. So we end up invoking the `FacetsWithEntryPoint` constructor instead of the `ExtraFacets` constructor. If you have any insight into this problem I'd love to hear it.

Reply:
Without this test:

```
>>> ExtraFacets.from_request
<bound method type.from_request of <class 'core.tests.test_lane.ExtraFacets'>>
```

With the test:

```
ExtraFacets._from_request
<bound method type._from_request of <class 'lane.FacetsWithEntryPoint'>>
```

Reply:
Maybe you could define a subclass of FacetsWithEntryPoint and override `_from_request` instead of changing the original method.

Reply:
Good idea, that's simpler too.